src/data_tools/data_transformer.py

```diff
@@ -0,0 +1,867 @@
+    
+    rows, cols, dim = image.shape
+    M = getRotationMatrix2D(center = (round(cols/2), round(rows/2)), angle=deg, scale=1)
+    ret = np.copy(image)
+    return cv2.warpAffine(ret, M, (int(cols),int(rows)))
+
+def shear_image(image, sh_x, sh_y, height=HEIGHT, width=WIDTH):
+    '''
+    Shears input image along x and y axes by specified amounts.
+
+    INPUTS
+    ------
+    image: Input image of dimension WIDTH x HEIGHT.
+    sh_x: shearing factor in x-axis.
+        must be less than ~ WIDTH / HEIGHT to stay within image frame
+        float in [0,WIDTH / HEIGHT)
+    sh_y: shearing factor in y-axis.
+        must be less than ~ HEIGHT / WIDTH to stay within image frame
+        float in [0,HEIGHT / WIDTH)
+    
+    NOTES
+    -----
+    Must be applied to labels as well.
+    '''
+    M = np.float32([[1, sh_x, 0],
+                    [sh_y, 1, 0],
+                    [0, 0, 1]])
+    inv = np.linalg.inv(M)
+    col = np.float32([[width], [height], [1]])
+    res = np.dot(inv,col)
+
+    new_size = resize_image(image, int(math.floor(res[0][0])), int(math.floor(res[1][0])), centerized=0, scaling_method="coordinates", height=height, width=width)
+    sheared_img = cv2.warpPerspective(new_size,M,(int(width),int(height)))
+
+    return sheared_img
+
+def apply_gaussian(image, kernel_x=10, kernel_y=10):
+    '''
+    Applies a gaussian blur to the input image.
+    
+    INPUTS
+    ------
+    image: Input image of dimension WIDTH x HEIGHT.
+    kernel_x and kernel_y: Determine the blurring. 1 is added to both after they are multipled by 2 to ensure the input into the cv2.GaussianBlur function is odd
+        positive integers
+        default values for both are 10.
+    
+    NOTES
+    -----
+    Does not affect labels.
+    '''
+    ret = np.copy(image)
+    return cv2.GaussianBlur(ret, (2*kernel_x + 1, 2*kernel_y + 1), 0)
+
+def colour(image, bf=1, gf=1, rf=1, mode="RGB"):
+    '''
+    Modifies colour channels of input image.
+
+    INPUTS
+    ------
+    image: Input image of dimensions WIDTH x HEIGHT.
+    bf: blue-factor to be multiplied to the blue channel of the image.
+        bf = 0 will remove all blue from the image
+        if bf increases the blue beyond 255, it gets capped at 255
+    gf: green-factor to be multiplied to the green channel of the image.
+        gf = 0 will remove all green from the image
+        if gf increases the green beyond 255, it gets capped at 255  
+    rf: red-factor to be multiplied to the red channel of the image.
+        rf = 0 will remove all red from the image
+        if rf increases the red beyond 255, it gets capped at 255  
+    mode: indicates if you want to modify the RGB channels of your image.
+        default modifies RGB channels
+        otherwise, applies grayscale to the image
+
+    NOTES
+    -----
+    Does not affect labels.
+
+    '''
+    if (mode != "RGB"):
+        return cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    else:
```

Comment:
Just a stylistic note, when you have a construct like this:

```python
if condition:
    return value
else:
    # More code
```

you can move the "# More code" out of the else statement and de-indent it.

```python
if condition:
    return value
# More code
```

They are exactly equivalent. Removing the extra indentation is nice because it's easier to read (the reader doesn't have to think so much about the first if-else statement. The first if statement that returns something is a if-guard